include/coap3/resource.h

```diff
@@ -22,6 +22,10 @@
 #define COAP_RESOURCE_CHECK_TIME 2
 #endif /* COAP_RESOURCE_CHECK_TIME */
 
+#ifndef COAP_RESOURCE_MAX_SUBSCRIBER
+#define COAP_RESOURCE_MAX_SUBSCRIBER 0 /* There is no maximum */
+#endif /* COAP_RESOURCE_MAX_SUBSCRIBER */
+
```

Comment:
In general, variable definitions such as this which applications use should be in the publicly available header files.  However, in  this case the definition is used to determine how libcoap is compiled which the application is unable to change, so should not be visible to the applications.

Therefore, variable definitions that are libcoap library specific should be in the appropriate i`nclude/coap3/coap_*_include.h` file - in this case `include/coap3/coap_resource_internal.h`.  Please move this definition there.